docs/docs/aztec/smart_contracts/injecting_data/index.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Injecting Data
+tags: [functions, oracles, capsules]
+---
+
+There are multiple ways to inject data into smart contracts.
+
+1. **Oracles** - fetching data from the outside world
+2. **Authwits (Authentication Witnesses)** - authorizing an arbitrary action (or piece of data)
+3. **Capsules** - local data storage in the PXE
+
+## Oracles
+
+In the world of EVM, you can read data directly from storage. However, when we are working with circuits for private execution, this becomes more tricky as you cannot just read the storage directly from your state tree, because there are only commitments (e.g. hashes) there. The pre-images (content) of your commitments need to be provided to the function to prove that you actually allowed to modify them.
+
+If we fetch the notes using an oracle call, we can keep the function signature independent of the underlying data and make it easier to use. See [oracles](./oracles.md) for more information.
```

Comment:
With this statement, it isn't clear to me what it means that "we can keep the function signature independent of the underlying data". 

This concept isn't only relevant for notes, right? I know it's liisted on the linked page, but it might provide clarity to list what things oracles are used for so readers get a sense (e.g. fetch notes from pxe, cross-chain messages, private logs). 

Also, noticing that [this](https://github.com/AztecProtocol/aztec-packages/pull/12043/files#diff-46d6a3f8f30f26c91ffef88e037901494adfdda156127400f4118d181dc8fab5R25) line is missing a word (maybe "data"?)